src/main/io/gps.c

```diff
@@ -1561,6 +2045,14 @@ static union {
     ubxNavSat_t sat;
     ubxCfgGnss_t gnss;
     ubxAck_t ack;
+    struct {
+        //we cant remove swVersion, because of memory spacing, but we dont need it
+        char swVersion[30];
+        char hwVersion[10];
+#ifdef USE_GPS_DEBUG
+        char extension[300];
+#endif
+    } ver;
```

Comment:
Do we really need this struct? I still don't understand why we need to reserve 300 bytes just to save verbose information about the Ublox version. Seems bloated and completely unnecessary.

Reply:
It is a bloat.
My understanding is that it provides the extra information that can be useful to determine the software version details needed to resolve some connection problems, eg when hardware doesn't match software.
It is only available in the `USE_GPS_DEBUG` custom build so doesn't cause bloat unless in a custom build.
As to the need for it, @freasy can explain better than I.

Reply:
I am talking about the whole `ver` struct, not just `extension[300]`

Reply:
It may be possible to hack it so that MSP (configurator) can request whole GPS rx buffer / response to specific command. Or simply switch to passthgrough mode and either implement it in Configurator or use Ublox app

Reply:
Interrogating the GPS using code in Configurator, and a GPS passthrough would be very nice, and would not require a custom firmware build.  Additionally we could provide tools in Configurator to permanently write settings to the module.

Reply:
@ctzsnooze : Reasonable settings shall stay in flash.

Reply:
This is in a union laid over the `_buffer`, correct? The largest element in that union is `uint8_t bytes[UBLOX_PAYLOAD_SIZE];` which is 392 bytes (at a 32 SAT MAX) bigger than the `ver` struct even with the 300 byte `extension` in it. Why is that array significant, we need to handle the actual payload sizes?

Having the `ver` struct lets us parse the UBX-MON-VER message payload. Not sure why it's not a separately defined type to be consistent with the other payload struct mappings? We have the `ubxMonVer_t` type in `gps.h`, but it's defined differently and not used here.

I think we do too much with the version info.

We only need the `hwVersion` from the UBX-MON-VER message to determine the module platform version (e.g., M8, M10), which is what we use for all of our command selection.

The `swVersion` field from the UBX-MON-VER message, and the other values embedded in the text `extension` field (such as PROTVER) are currently for info only, displayed by the cli commands. Maybe we think these are useful, I'd argue not so much--if we're troubleshooting someone's module at that level we get them to go into gpspassthrough and use u-center. For BF troubleshooting, we only need to know what platform version BF thinks the module is.

We take the hwVersion, which is a value such as `00080000` and map that to the platform version (e.g., M8). So, we need to parse the command for the hwVersion field, but the parsing case could just ignore the rest of the message.

The UBX-MON-VER parsing code in `UBLOX_parse_gps( )` could extract the hwVersion and do the mapping right there, setting the various `ubloxM7orAbove` flags we use. It does that now, we can just strip out the unnecessary parsing.

cli status could use those flags rather printing the actual hwVersion field. We wouldn't need to save any of the raw data from the command.

I searched and didn't see anything other than `cli.c` using the `gpsData.monVer` data. Unless I missed something...

Reply:
Resolved, I think, by your recent commit which now reports the Hardware version in the CLI, and removes the old software version and extension code, to be discussed in a later PR.

Reply:
> @ZzyzxTek:
> This is in a union laid over the _buffer, correct? The largest element in that union is uint8_t bytes[UBLOX_PAYLOAD_SIZE]; which is 392 bytes (at a 32 SAT MAX) bigger than the ver struct even with the 300 byte extension in it.

Good point. In that case the `ver` struct doesn't waste any additional memory. I really dropped the ball on this PR review. I overlooked or mixed up several things in a hurry...